include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,32 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void concatenate(const std::vector<TensorListShape<sample_ndim>> &tlss) {
```

Comment:
1. Concatenation of TensorShapes is a free function `shape_cat`. This one should be a free function, too, since the original content of the list is discarded.
2. Use `span` instead of `vector` - this will enable more flexible usage.

I think that it might be better to have an `append` function that just appends one tensor list to an existing one.
The caller will be able to iterate over a collection of their choice.

Reply:
As a small optimization, I went for both. The thing is, that in case of iterating over a collection, more than one `shapes` reallocation might occur. On the other hand, `make_span` accepts collection of user's choice.